airtable/emails.js

```diff
@@ -5,165 +5,146 @@
+    const volunteers = delivery.getCellValue("Assigned volunteer");
+    if (volunteers && volunteers.length > 0) {
+        volunteer = findById(allVolunteers, volunteers[0].id);
+    }
+    
+    for (let r of templateVarsConfig) {
+        if (r.getCellValue("Airtable table").name == "Deliveries") {
+            output.delivery[r.getCellValue("sendgrid")] = delivery.getCellValue(r.getCellValue("Airtable Field Name"));
+        } else if (r.getCellValue("Airtable table").name == "Volunteers") {
+            if (volunteer) {
+                output.volunteer[r.getCellValue("sendgrid")] = volunteer.getCellValue(r.getCellValue("Airtable Field Name"));
+            }
+        } else {
+            console.log("Configuration error, make sure all template variables have either Deliveries or Volunteers as their Airtable table");
         }
-    } else {
-        volunteerData = null;
     }
 
-    return {
-        delivery: {
-            name: delivery.getCellValue("Recipient Name"),
-            address: delivery.getCellValue("Address"),
-            details: delivery.getCellValue("Delivery Request Details"),
-            phone: delivery.getCellValue("Phone Number"),
-            email: delivery.getCellValue("Email"),
-            instructions: delivery.getCellValue("Delivery Instructions"),
-            // we need to have a boolean value for each grocery store,
-            // in order to branch conditionally on the store in the email
-            grocery_store: {
-                shoprite: delivery.getCellValue("Grocery Store Name").name === "ShopRite, Little Falls",
-                golden_mango: delivery.getCellValue("Grocery Store Name").name === "Golden Mango, Paterson",
-                pay_later: delivery.getCellValue("Payment Method").name !== "(Preferred) Call ahead to the store and pay over the phone."
-            }
-        },
-        volunteer: volunteerData
-    };
+    return output;
 }
 
-//==========================================================
-//==========================================================
-// Standard script begins here --
-// You shouldn't need to change anything below this line.
-//==========================================================
-
-// Define helper functions
-//==========================================================
-
-const messagesTable = base.getTable("Messages")
-const volunteersTable = base.getTable("Volunteers")
-const deliveriesTable = base.getTable("Deliveries")
-const allMessages = (await messagesTable.selectRecordsAsync()).records;
-const allVolunteers = (await volunteersTable.selectRecordsAsync()).records;
-const allDeliveries = (await deliveriesTable.selectRecordsAsync()).records;
+async function formatSengridMessage(delivery, messageType) {
+    const sendgridData = await getEmailInfoForDelivery(delivery);
+
+    const audience = EMAIL_TYPES[messageType]["audience"];
+    var toEmail;
+    if (audience === "volunteer") {
+        toEmail = sendgridData.volunteer?.email;
```

Comment:
Wow, `?.` is new to me -- so useful!

Reply:
I guess not quite -- the bugfix itself is more inside `getEmailInfoForDelivery`

Reply:
Yep, a combination of having the volunteer key in sendgridData default to {} rather than null, plus this fix, makes that particular bug go away :)